Authenticator/Source/TableDiff.swift

```diff
@@ -80,7 +80,7 @@ private func changes<T>(from oldItems: [T], to newItems: [T], hasSameIdentity: (
             var y = x - k
             while x < oldItems.count && y < newItems.count
                 && hasSameIdentity(oldItems[x], newItems[y]) {
-                    if !isEqual(oldItems[x], newItems[y]) {
+                    if !isEqual(oldItems[x], newItems[y]) && (x != y) {
```

Comment:
II was experiencing crashes on my branch when sorting the last item upwards. Remove this additional logic check and you'll see it. I can post the error trace but it had to do with a single row having more than one operation.

I believe the underlying problem may have to do with the map/filter of the filtered tokens but never got to the bottom of it.